hal/src/hal-sdcard.adb

```diff
@@ -144,12 +151,29 @@ package body HAL.SDCard is
          Info.Card_Type := STD_Capacity_SD_Card_V1_1;
       end if;
 
-      --  ACMD41: SD_SEND_OP_COND (no crc check)
-      --  Arg: HCS=1, XPC=0, S18R=0
       for I in 1 .. 5 loop
          delay until Clock + Milliseconds (200);
-         Send_ACmd (Driver, SD_App_Send_Op_Cond, 0, 16#40ff_0000#, Status);
 
+         --  CMD55: APP_CMD
+         --  This is done manually to handle error (this command is not
+         --  supported by mmc).
+         Send_Cmd (Driver, Cmd_Desc (App_Cmd), 0, Status);
+         if Status /= OK then
+            if Status = Command_Timeout_Error
+              and then I = 1
+              and then Info.Card_Type = STD_Capacity_SD_Card_V1_1
+            then
+               --  Not an SDCard.  Suppose MMC.
+               Info.Card_Type := Multimedia_Card;
+               exit;
+            end if;
+            return;
+         end if;
+
+         --  ACMD41: SD_SEND_OP_COND (no crc check)
+         --  Arg: HCS=1, XPC=0, S18R=0
+         Send_Cmd
+           (Driver, Acmd_Desc (SD_App_Send_Op_Cond), 16#40ff_0000#, Status);
          if Status /= OK then
             Put_Line ("send_op_cond failed");
```

Comment:
Debug messages shouldn't stay here. At least they must be disabled by default.

(Commenting on a line of code that is not removed by a following patch)